engine/planner.go

```diff
@@ -268,8 +270,29 @@ func (p *planner) listBusy(ctx context.Context) (map[string]struct{}, error) {
 // helper function returns true if the os, arch, variant
 // and kernel match the stage.
 func (p *planner) match(stage *drone.Stage) bool {
+	labelMatch := true
+
+	if len(p.labels) > 0 || len(stage.Labels) > 0 {
+		labelMatch = p.matchLabel(stage)
+	}
+
 	return stage.OS == p.os &&
 		stage.Arch == p.arch &&
 		stage.Variant == p.version &&
-		stage.Kernel == p.kernel
+		stage.Kernel == p.kernel &&
+		labelMatch
```

Comment:
We could match with `reflect.DeepEqual(stage.Labels, p.labels)` instead

Reply:
this probably comes down to personal preference, but I prefer the current approach of using a typed function.